src/test/java/uk/gov/companieshouse/api/accounts/validation/DebtorsValidatorTest.java

```diff
@@ -0,0 +1,287 @@
+        previousDebtors.setPrepaymentsAndAccruedIncome(4L);
+        previousDebtors.setGreaterThanOneYear(6L);
+        previousDebtors.setOtherDebtors(8L);
+        previousDebtors.setTotal(20L);
+
+        debtors.setPreviousPeriod(previousDebtors);
+
+        when(transaction.getCompanyNumber()).thenReturn("12345");
+        when(environmentReader.getMandatoryString(any())).thenReturn("http://test-url");
+        when(restTemplate.getForObject(anyString(), any(Class.class))).thenReturn(companyProfile);
+
+        validator.isMultipleYearFiler(transaction);
+        errors = validator.validateDebtors(debtors, transaction);
+
+        assertFalse(errors.hasErrors());
+
+    }
+
+    @Test
+    @DisplayName("Tests the validation fails on single year filer filing previous period")
+    public void tesInvalidMultipleYearDebtorsNote() throws Throwable {
+
+        CompanyProfile companyProfile = new CompanyProfile();
+
+        addValidCurrentDebtors();
+
+        PreviousPeriod previousDebtors = new PreviousPeriod();
+        previousDebtors.setTradeDebtors(2L);
+        previousDebtors.setTotal(20L);
+
+        debtors.setPreviousPeriod(previousDebtors);
+
+        when(transaction.getCompanyNumber()).thenReturn("12345");
+        when(environmentReader.getMandatoryString(any())).thenReturn("http://test-url");
+        when(restTemplate.getForObject(anyString(), any(Class.class))).thenReturn(companyProfile);
+
+        ReflectionTestUtils.setField(validator, "inconsistentData", "inconsistent_data");
+
+        validator.isMultipleYearFiler(transaction);
+        errors = validator.validateDebtors(debtors, transaction);
+
+        assertTrue(errors.hasErrors());
+
+        assertTrue(errors.containsError(
+            new Error("inconsistent_data", PREVIOUS_TRADE_DEBTORS,
+                LocationType.JSON_PATH.getValue(),
+                ErrorType.VALIDATION.getType())));
+
+        assertTrue(errors.containsError(
+            new Error("inconsistent_data", PREVIOUS_TOTAL_PATH,
+                LocationType.JSON_PATH.getValue(),
+                ErrorType.VALIDATION.getType())));
+
+    }
+
+    @Test
+    @DisplayName("Tests the validation fails on previous period incorrect total")
+    public void testIncorrectPreviousDebtorsTotal() throws Throwable {
+
+        CompanyProfile companyProfile = addMultipleYearFilingCompany();
+
+        addValidCurrentDebtors();
+
+        PreviousPeriod previousDebtors = new PreviousPeriod();
+        previousDebtors.setTradeDebtors(2L);
+        previousDebtors.setTotal(20L);
+
+        debtors.setPreviousPeriod(previousDebtors);
+
+        when(transaction.getCompanyNumber()).thenReturn("12345");
+        when(environmentReader.getMandatoryString(any())).thenReturn("http://test-url");
+        when(restTemplate.getForObject(anyString(), any(Class.class))).thenReturn(companyProfile);
+
+        ReflectionTestUtils.setField(validator, "incorrectTotal", "incorrect_total");
```

Comment:
`incorrectTotal` and `incorrect_total` are used a few times and could be extracted into constants:

`private static final String INCORRECT_TOTAL_NAME = "incorrectTotal";`
`private static final String INCORRECT_TOTAL_VALUE = "incorrect_total";`